ss/pebbledb/db.go

```diff
@@ -184,13 +185,88 @@ func (db *Database) ApplyChangeset(version int64, cs *proto.NamedChangeSet) erro
 	return b.Write()
 }
 
-// Prune for the PebbleDB SS backend is currently not supported. It seems the only
-// reliable way to prune is to iterate over the desired domain and either manually
-// tombstone or delete. Either way, the operation would be timely.
-//
-// See: https://github.com/cockroachdb/cockroach/blob/33623e3ee420174a4fd3226d1284b03f0e3caaac/pkg/storage/mvcc.go#L3182
+// Prune attempts to prune all versions up to and including the current version
+// Get the range of keys, manually iterate over them and delete them
 func (db *Database) Prune(version int64) error {
```

Comment:
Quick question. How do you ensure you return `<nil, nil>` for queries that are less than or equal to a pruned height? 

For example, RocksDB throws an `smaller than full_history_ts_low` error which we can easily catch and return `<nil, nil>` upon. But for PebbleDB, how do you do this?

You would need to manually store the latest height pruned I think.

Reply:
This is a good question, but shouldn't this happen automatically when you call Get or Has and since it couldn't find the key it will just return nil?

Reply:
No, you'll still have the key. Take your example from [here](https://github.com/sei-protocol/sei-db/pull/27#discussion_r1391561344). If I prune at height=150 query for k1 at height=160, I should get v1 as expected. But if I query for k1 and height=150 (or less), I should get `<nil, nil>`.

Reply:
Ooh! I think you are right, yeah we need to store a earliest version metadata for pruned height

Reply:
Yup. Just store the height when `Prune` is called and ensure any read respects this value.

Reply:
@alexanderbez this was updated on latest commit. we add a `earliestVersion` both stored in the db and loaded into memory. We do the same operations as in the rocks example